chore(deps): bump openjd-adaptor-runtime to 0.9.*

jericht opened the pull request:
### What was the problem/requirement? (What/Why)
We should upgrade `openjd-adaptor-runtime` to the latest released version to get all the latest fixes and improvements.

### What was the solution? (How)
Upgrade `openjd-adaptor-runtime` to 0.9.*

### What is the impact of this change?
We have the latest adaptor runtime changes

### How was this change tested?

#### Manual testing
- Submitted a Maya job with the developer override adaptor option enabled, verified the job succeed, downloaded the output and ensured it looked correct. Used the "Sophie" scene from https://help.autodesk.com/view/ARNOL/ENU/?guid=arnold_for_maya_tutorials_am_Learning_Scenes_html

#### Integ test result

I only ran the Maya render scene, could not get the Redshift one to work even after installing Redshift and V-Ray for Maya 2025

```
$ hatch run integ:test_adaptors_maya
================================================================================== test session starts ===================================================================================
platform darwin -- Python 3.11.4, pytest-8.4.1, pluggy-1.6.0 -- /Applications/Autodesk/maya2025/Maya.app/Contents/bin/mayapy
cachedir: .pytest_cache
rootdir: /Volumes/workplace/github/deadline-cloud-for-maya
configfile: pyproject.toml
plugins: flaky-3.8.1, xdist-3.8.0, cov-6.2.1
1 worker [1 item]      
scheduling tests via LoadScheduling

test/integ/test_maya_adaptors.py::TestAdaptors::test_minimal_scene_adaptor 
[gw0] [100%] PASSED test/integ/test_maya_adaptors.py::TestAdaptors::test_minimal_scene_adaptor 

==================================================================================== warnings summary ====================================================================================
test/integ/test_maya_adaptors.py:27
  /Volumes/workplace/github/deadline-cloud-for-maya/test/integ/test_maya_adaptors.py:27: PytestUnknownMarkWarning: Unknown pytest.mark.maya_renderer - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.maya_renderer

test/integ/test_maya_adaptors.py:51
  /Volumes/workplace/github/deadline-cloud-for-maya/test/integ/test_maya_adaptors.py:51: PytestUnknownMarkWarning: Unknown pytest.mark.redshift_renderer - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.redshift_renderer

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
===Flaky Test Report===

test_minimal_scene_adaptor passed 1 out of the required 1 times. Success!

===End Flaky Test Report===
================================================================================== slowest 5 durations ===================================================================================
33.77s call     test/integ/test_maya_adaptors.py::TestAdaptors::test_minimal_scene_adaptor
0.00s setup    test/integ/test_maya_adaptors.py::TestAdaptors::test_minimal_scene_adaptor
0.00s teardown test/integ/test_maya_adaptors.py::TestAdaptors::test_minimal_scene_adaptor
============================================================================= 1 passed, 2 warnings in 35.34s =============================================================================
```

#### Installer test result

```
$ hatch run test-installer
================================================================================== test session starts ===================================================================================
platform darwin -- Python 3.9.6, pytest-8.4.1, pluggy-1.6.0 -- /Users/jericht/Library/Application Support/hatch/env/virtual/deadline-cloud-for-maya/vTaURC9A/deadline-cloud-for-maya/bin/python
cachedir: .pytest_cache
rootdir: /Volumes/workplace/github/deadline-cloud-for-maya
configfile: pyproject.toml
plugins: xdist-3.8.0, cov-6.2.1
12 workers [13 items]     
scheduling tests via LoadScheduling

test/installer/test_installer.py::TestVerifySigning::test_linux_signing 
test/installer/test_installer.py::TestWindows::test_system_permissions 
test/installer/test_installer.py::TestUserInstall::test_install 
test/installer/test_installer.py::TestLinuxAndMacOS::test_system_permissions 
test/installer/test_installer.py::test_did_not_build_with_evaluation_mode 
test/installer/test_installer.py::TestWindows::test_user_permissions 
test/installer/test_installer.py::test_default_location 
test/installer/test_installer.py::TestSystemInstall::test_uninstall 
[gw9] [  7%] SKIPPED test/installer/test_installer.py::TestSystemInstall::test_uninstall 
[gw5] [ 15%] SKIPPED test/installer/test_installer.py::TestWindows::test_system_permissions 
[gw3] [ 23%] SKIPPED test/installer/test_installer.py::TestLinuxAndMacOS::test_system_permissions 
test/installer/test_installer.py::TestVerifySigning::test_windows_signing 
[gw10] [ 30%] SKIPPED test/installer/test_installer.py::TestVerifySigning::test_windows_signing 
[gw11] [ 38%] SKIPPED test/installer/test_installer.py::TestVerifySigning::test_linux_signing 
test/installer/test_installer.py::TestLinuxAndMacOS::test_user_permissions 
[gw1] [ 46%] SKIPPED test/installer/test_installer.py::test_did_not_build_with_evaluation_mode 
[gw4] [ 53%] SKIPPED test/installer/test_installer.py::TestWindows::test_user_permissions 
test/installer/test_installer.py::TestUserInstall::test_uninstall 
test/installer/test_installer.py::TestSystemInstall::test_install 
[gw8] [ 61%] SKIPPED test/installer/test_installer.py::TestSystemInstall::test_install 
[gw0] [ 69%] PASSED test/installer/test_installer.py::test_default_location 
test/installer/test_installer.py::TestVerifySigning::test_macos_signing 
[gw0] [ 76%] SKIPPED test/installer/test_installer.py::TestVerifySigning::test_macos_signing 
[gw6] [ 84%] PASSED test/installer/test_installer.py::TestUserInstall::test_install 
[gw2] [ 92%] PASSED test/installer/test_installer.py::TestLinuxAndMacOS::test_user_permissions 
[gw7] [100%] PASSED test/installer/test_installer.py::TestUserInstall::test_uninstall 

================================================================================== slowest 5 durations ===================================================================================
11.82s setup    test/installer/test_installer.py::TestLinuxAndMacOS::test_user_permissions
11.77s setup    test/installer/test_installer.py::TestUserInstall::test_install
11.03s setup    test/installer/test_installer.py::TestUserInstall::test_uninstall
3.12s call     test/installer/test_installer.py::TestUserInstall::test_uninstall
2.64s call     test/installer/test_installer.py::test_default_location
============================================================================= 4 passed, 9 skipped in 16.21s ==============================================================================
```

#### Did you run the "Job Bundle Output Tests"? If not, why not? If so, paste the test results here.

No, the adaptor runtime is not used by the submitter.

### Was this change documented?

No

### Is this a breaking change?

No

----

*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.*

jericht committed:
chore(deps): bump openjd-adaptor-runtime to 0.9.*

Signed-off-by: Jericho Tolentino <68654047+jericht@users.noreply.github.com>

kenyrish-amazon reviewed (commented):
Theres a TODO for when we bump this to 0.9.*, can we do this as well?
https://github.com/aws-deadline/deadline-cloud-for-maya/blob/85d880c3915bfe0f566c23c77725692c2bfa83cf/src/deadline/maya_adaptor/MayaClient/maya_client.py#L17

leongdl commented on DEVELOPMENT.md:
````diff
@@ -285,18 +285,47 @@ or modifying functionality, then you will want to be writing one or more integ t
    /Applications/Autodesk/maya<version>/Maya.app/Contents/MacOS
    ```
-2. Use hatch to run all adaptor tests:
+4. Navigate to your local checkout of this repository:
````
I think the steps 4-8 can be done as 1 step by calling `hatch run install`

jericht replied:
I just tried on a fresh Maya 2025 install and it doesn't work:

```
$ hatch run integ:test_submitters
/Applications/Autodesk/maya2025/Maya.app/Contents/bin/mayapy: No module named pytest
```

I took a look (see [code](https://github.com/aws-deadline/deadline-cloud-for-maya/blob/85d880c3915bfe0f566c23c77725692c2bfa83cf/scripts/install_dev_submitter.py)) and the command just creates a `plugin_env` with all the required submitter components installed into it that you can set your `MAYA_MODULE_PATH` to so that Maya loads the submitter from there. It doesn't look like it installs anything into Maya's Python environment, which is what the integ tests are running by calling `mayapy` (see [hatch.toml](https://github.com/aws-deadline/deadline-cloud-for-maya/blob/85d880c3915bfe0f566c23c77725692c2bfa83cf/hatch.toml#L44-L49))

leongdl replied:
OK thanks for checking

leongdl commented:
FAILED test/unit/deadline_adaptor_for_maya/MayaClient/test_client.py::TestMayaClient::test_mayaclient - AttributeError: <module 'deadline.maya_adaptor.MayaClient.maya_client' from '/Users/runner/work/deadline-cloud-for-maya/deadline-cloud-for-maya/src/deadline/maya_adaptor/MayaClient/maya_client.py'> does not have the attribute 'HTTPClientInterface'
FAILED test/unit/deadline_adaptor_for_maya/MayaClient/test_client.py::TestMayaClient::test_main - AttributeError: <module 'deadline.maya_adaptor.MayaClient.maya_client' from '/Users/runner/work/deadline-cloud-for-maya/deadline-cloud-for-maya/src/deadline/maya_adaptor/MayaClient/maya_client.py'> does not have the attribute 'HTTPClientInterface'

jericht commented:
> FAILED test/unit/deadline_adaptor_for_maya/MayaClient/test_client.py::TestMayaClient::test_mayaclient - AttributeError: <module 'deadline.maya_adaptor.MayaClient.maya_client' from '/Users/runner/work/deadline-cloud-for-maya/deadline-cloud-for-maya/src/deadline/maya_adaptor/MayaClient/maya_client.py'> does not have the attribute 'HTTPClientInterface' FAILED test/unit/deadline_adaptor_for_maya/MayaClient/test_client.py::TestMayaClient::test_main - AttributeError: <module 'deadline.maya_adaptor.MayaClient.maya_client' from '/Users/runner/work/deadline-cloud-for-maya/deadline-cloud-for-maya/src/deadline/maya_adaptor/MayaClient/maya_client.py'> does not have the attribute 'HTTPClientInterface'

Fixed. @leongdl 



> Theres a TODO for when we bump this to 0.9.*, can we do this as well?
> 
> https://github.com/aws-deadline/deadline-cloud-for-maya/blob/85d880c3915bfe0f566c23c77725692c2bfa83cf/src/deadline/maya_adaptor/MayaClient/maya_client.py#L17

Thanks, updated and retested it on Windows, verified render is working. @kenyrish-amazon 

Side note, I think something is up with the GitHub Mac runners and this PR isn't getting any
